### ISSUE TITLE
Expected UI issues with Wagtail 3.0 release

### ISSUE THREAD

**Issue by thibaudcolas** [torchbox/wagtailmedia#157]
The [Wagtail 3.0 first release candidate](https://github.com/wagtail/wagtail/discussions/8362) is out. There are large UI changes in this release, for which we have [reviewed expected breakage in third-party UI customisations](https://github.com/wagtail/wagtail/discussions/7739#discussioncomment-2615089).

This is beyond what we do with our normal breaking changes policy, since the majority of those changes are on parts of Wagtail that haven’t been publicly supported / documented in any way. To make sure this goes smoothly anyway, I’m here to provide an advance notice of what we’re aware of with this specific package 🙂 

In the case of wagtailmedia, the only issue is reliance on Wagtail’s internal tabs implementation:

```text
torchbox/wagtailmedia/src/wagtailmedia/templates/wagtailmedia/chooser/chooser.html
15:    <ul class="tab-nav merged" data-tab-nav>
```

Here is our [guidance on replacing Bootstrap tabs](https://github.com/wagtail/wagtail/discussions/7739#user-content-2022-04-22-breaking-changes-in-ui-code-bootstrap-tabs), and our [proposal for a tabs component API](https://github.com/wagtail/wagtail/issues/8408).

**Comment by th3hamm0r** [comment]
Is there any timeline when this will be fixed? Just found out for an already upgraded project, that the media chooser is kind of broken :(

**Comment by zerolab** [comment]
@th3hamm0r I am aiming to get this sorted this week.

**Comment by th3hamm0r** [comment]
@zerolab that would be awesome!
I've looked into the templates, but I'm actually not an expert here. For example, I don't really know, how you will solve the compatibility with both wagtail 2.x and 3.x. If I succeed to temporarily fix the template for our project, I may be able to help you with some code (but I cannot promise that^^).

**Comment by zerolab** [comment]
One option is to have 2 different templates. The other is to pass a variable in the template context and vary the markup based on that

**Comment by th3hamm0r** [comment]
I assume the js needs adaptions too, because clicking the tabs don't work with only the template changes (based on image/document chooser). It was probably naive of me to think that this is only a template change. So atm I can only support you by testing your changes, sry.

**Comment by zerolab** [comment]
@th3hamm0r https://github.com/wagtail/wagtail/discussions/7739#user-content-2022-04-22-breaking-changes-in-ui-code-bootstrap-tabs mentions only the markup needs changing 🤔

**Comment by th3hamm0r** [comment]
@zerolab Yes, it looks simple, but I think this only works if the tabs are already in the DOM, because then this `initTabs` call does the initializing:
https://github.com/wagtail/wagtail/blob/6343a9558cd33625d7d0338f1ff703ffd976e027/client/src/entrypoints/admin/wagtailadmin.js#L29

That's probably the reason, why the core modals, e.g. for images, trigger that function again:
https://github.com/wagtail/wagtail/blob/6343a9558cd33625d7d0338f1ff703ffd976e027/client/src/entrypoints/images/image-chooser-modal.js#L151

On a quick search I haven't found any public API to trigger `initTabs`.

**Comment by th3hamm0r** [comment]
Fyi: this is the change template I've tried:
https://gist.github.com/th3hamm0r/d47e4085d6b748f20b0e9704342075bb

**Comment by zerolab** [comment]
@th3hamm0r got a PR ready - #161

**Comment by th3hamm0r** [comment]
@zerolab perfect, thanks, I think I will be able to test it tomorrow
